r1cs-std/src/fields/mod.rs

```diff
@@ -12,7 +15,15 @@ pub mod fp4;
 pub mod fp6_2over3;
 pub mod fp6_3over2;
 
-pub trait FieldGadget<F: Field, ConstraintF: Field>:
+use crate::fields::fp::FpGadget;
+pub trait ToConstraintFieldGadget<ConstraintF: PrimeField> {
+    fn to_field_gadgets<CS: ConstraintSystem<ConstraintF>>(
```

Comment:
it may be clearer to name this `to_constraint_field`? So that a reader understands that `x.to_field_gadgets` actually means that you're getting some field elements in the constraint field.

Reply:
Agree. I will do the renaming in a bit and push a commit.